src/controllers/accomController.js

```diff
@@ -0,0 +1,37 @@
+import * as ApplicationError from '../utils/errors/applicationsErrors';
+import { deleteOrCreate } from '../services/likeService';
+import { Facility,Like } from '../database/models';
+
+const likeOrUnlike = async (req,res)=>{
+    try{
+    const {facility_id: facilityId} = req.params;
+    const {id:userId} = req.user;
+    const facilityExist = await Facility.findOne({where: {id: facilityId}})
+    if(!facilityExist){
+        return ApplicationError.notFoundError("This facility does not exist", res);
+    }
+    const {created} = await deleteOrCreate(Like, {facility_id: facilityId, user_id: userId},)
+    if(created){
+        return res.status(201).json({status: 201, message: `You have liked facility ${facilityId}`})
+    }
+    return res.status(200).json({status: 200, message: `You have unliked facility ${facilityId}`}) 
+}
+catch(error){
+    return ApplicationError.internalServerError(error.message, res)
+}
+
+}
+const getLikes = async (req, res) =>{
+    try{
+    const {facility_id: facilityId} = req.params;
+    const facilityLikes = await Like.findAll({where: {facility_id: facilityId}})
```

Comment:
Count the number of likes instead, This will require more unnecessary manipulations